plugins/source/digitalocean/codegen/resources.go

```diff
@@ -0,0 +1,398 @@
+			},
+		},
+		SkipFields: []string{"NeighborId"},
+	},
+	{
+		Service: "firewalls",
+		Struct:  godo.Firewall{},
+		ExtraColumns: []codegen.ColumnDefinition{
+			{
+				Name:     "droplet_ids",
+				Type:     schema.TypeIntArray,
+				Resolver: `schema.PathResolver("DropletIDs")`,
+			},
+			{
+				Name:     "id",
+				Type:     schema.TypeString,
+				Resolver: `schema.PathResolver("ID")`,
+				Options:  schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+		},
+		SkipFields: []string{"DropletIDs", "ID"},
+	},
+
+	{
+		Service: "floating_ips",
+		Struct:  godo.FloatingIP{},
+		ExtraColumns: []codegen.ColumnDefinition{
+			{
+				Name:     "ip",
+				Type:     schema.TypeString,
+				Resolver: `schema.PathResolver("IP")`,
+				Options:  schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+		},
+		SkipFields: []string{"IP"},
+	},
+	{
+		Service: "images",
+		Struct:  godo.Image{},
+		ExtraColumns: []codegen.ColumnDefinition{
+			{
+				Name:     "id",
+				Type:     schema.TypeInt,
+				Resolver: `schema.PathResolver("ID")`,
+				Options:  schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+		},
+		SkipFields: []string{"ID"},
+	},
+	{
+		Service: "keys",
+		Struct:  godo.Key{},
+		ExtraColumns: []codegen.ColumnDefinition{
+			{
+				Name:     "id",
+				Type:     schema.TypeInt,
+				Resolver: `schema.PathResolver("ID")`,
+				Options:  schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+		},
+		SkipFields: []string{"ID"},
+	},
+	{
+		Service:   "projects",
+		Struct:    godo.Project{},
+		Relations: []string{"Resources()"},
+		ExtraColumns: []codegen.ColumnDefinition{
+			{
+				Name:     "id",
+				Type:     schema.TypeString,
+				Resolver: `schema.PathResolver("ID")`,
+				Options:  schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+		},
+		SkipFields: []string{"ID"},
+	},
+	{
+		Service:    "projects",
+		SubService: "resources",
+		Struct:     godo.ProjectResource{},
+		ExtraColumns: []codegen.ColumnDefinition{
+			{
+				Name:     "urn",
+				Type:     schema.TypeString,
+				Resolver: `schema.PathResolver("URN")`,
+				Options:  schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+		},
+		SkipFields: []string{"URN"},
+	},
+	{
+		Service:   "registries",
+		Struct:    &godo.Registry{},
+		Relations: []string{"Repositories()"},
+		ExtraColumns: []codegen.ColumnDefinition{
+			{
+				Name:     "name",
+				Type:     schema.TypeString,
+				Resolver: `schema.PathResolver("Name")`,
+				Options:  schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+		},
+		SkipFields: []string{"Name"},
+	},
+	{
+		Service:    "registries",
+		SubService: "repositories",
+		Struct:     &godo.Repository{},
+		ExtraColumns: []codegen.ColumnDefinition{
+			{
+				Name:     "name",
+				Type:     schema.TypeString,
+				Resolver: `schema.PathResolver("Name")`,
+				Options:  schema.ColumnCreationOptions{PrimaryKey: true},
+			},
+		},
+		SkipFields: []string{"Name"},
+	},
+	{
+		Service: "sizes",
+		Struct:  godo.Size{},
```

Comment:
Is there a reasonable primary key we can use here? (It's okay if not, but better if there is)